source/auth_credentials.c

```diff
@@ -340,3 +332,59 @@ PyObject *aws_py_credentials_provider_new_static(PyObject *self, PyObject *args)
     Py_DECREF(capsule);
     return NULL;
 }
+
+PyObject *aws_py_credentials_provider_new_profile(PyObject *self, PyObject *args) {
+    (void)self;
+    struct aws_allocator *allocator = aws_py_get_allocator();
+
+    PyObject *bootstrap_py;
+    struct aws_byte_cursor profile_name;
+    struct aws_byte_cursor config_file_name;
+    struct aws_byte_cursor credentials_file_name;
+
+    if (!PyArg_ParseTuple(
+            args,
+            "Os#s#s#",
```

Comment:
you can do `z#` instead of `s#`
then if `None` comes from Python, then the ptr will be set NULL and the len will be set 0
it's quite convenient